wallet/src/actors/app/methods.rs

```diff
@@ -545,6 +548,27 @@ impl App {
         Ok(())
     }
 
+    /// Handle node status notifications received from a Witnet node.
+    pub fn handle_node_status_notification(&mut self, value: types::Json) -> Result<()> {
+        let status = serde_json::from_value::<types::StateMachine>(value).map_err(node_error)?;
+
+        // This iterator is collected early so as to free the immutable reference to `self`.
+        let wallets: Vec<types::SessionWallet> = self
+            .state
+            .wallets
+            .iter()
+            .map(|(_, wallet)| wallet.clone())
+            .collect();
+
+        for wallet in &wallets {
+            let sink = self.state.get_sink(&wallet.session_id);
+            self.handle_node_status_in_worker(&status, wallet, sink.clone());
+            self.state.update_node_state(status);
+        }
```

Comment:
This can be moved outside of the for loop

```suggestion
        }
        self.state.update_node_state(status);
```